flag_engine/features/models.py

```diff
@@ -19,41 +21,77 @@ def __hash__(self):
         return hash(self.id)
 
 
-@dataclass
-class MultivariateFeatureOptionModel:
+class MultivariateFeatureOptionModel(BaseModel):
     value: typing.Any
     id: int = None
 
 
-@dataclass
-class MultivariateFeatureStateValueModel:
+class MultivariateFeatureStateValueModel(BaseModel):
     multivariate_feature_option: MultivariateFeatureOptionModel
-    percentage_allocation: float
+    percentage_allocation: confloat(ge=0, le=100)
     id: int = None
-    mv_fs_value_uuid: str = field(default_factory=uuid.uuid4)
+    mv_fs_value_uuid: UUID4 = Field(default_factory=uuid.uuid4)
 
 
-@dataclass
-class FeatureSegmentModel:
+class FeatureSegmentModel(BaseModel):
     priority: int = None
 
 
-@dataclass
-class FeatureStateModel:
+class MultivariateFeatureStateValueList(
+    BaseCollectionModel[MultivariateFeatureStateValueModel]
+):
+    @classmethod
+    def __get_validators__(
+        cls,
+    ) -> typing.Generator[typing.Callable[..., typing.Any], None, None]:
+        yield cls.validate
+        yield cls.ensure_correct_percentage_allocations
+
+    @classmethod
+    def ensure_correct_percentage_allocations(
+        cls,
+        value: typing.List[MultivariateFeatureStateValueModel],
+    ) -> typing.List[MultivariateFeatureStateValueModel]:
+        if (
+            sum(
+                multivariate_feature_state.percentage_allocation
+                for multivariate_feature_state in value
+            )
+            > 100
+        ):
+            raise InvalidPercentageAllocation(
+                "Total percentage allocation for feature must be less or equal to 100 percent"
+            )
+        return value
+
+    def append(
+        self,
+        multivariate_feature_state_value: MultivariateFeatureStateValueModel,
+    ) -> None:
+        self.ensure_correct_percentage_allocations(
+            [*self, multivariate_feature_state_value],
+        )
+        super().append(multivariate_feature_state_value)
+
+
+class FeatureStateModel(BaseModel):
+    class Config:
+        validate_assignment: bool = True
+
     feature: FeatureModel
     enabled: bool
     django_id: int = None
     feature_segment: FeatureSegmentModel = None
-    featurestate_uuid: str = field(default_factory=uuid.uuid4)
-    feature_state_value: typing.Any = field(default=None, init=False)
```

Comment:
Just to confirm here, as far as I'm aware the `init=False` argument to the `field()` function means that the value is not a valid argument to `__init__()`, allowing us to force users to set the value using the `set_value()` method. I'm not _entirely_ sure why we did this but I want to make sure that we're not losing/breaking any functionality here?

Reply:
@khvn26 I think this question is still outstanding here?

Reply:
We're not breaking anything, only adding an ability to set `feature_state_value` via `__init__` as a usual dataclass attribute.

The `set_value` method is kept as it was for backwards compatibility.

Reply:
The `set_value` method is a legacy interface initially conceived to get the feature state value that was private.
To simplify serialization it was made public, and now this method is redundant.
It's still being kept for backwards compatibility but we'll steer clear of using it in the future and deprecate it eventually.